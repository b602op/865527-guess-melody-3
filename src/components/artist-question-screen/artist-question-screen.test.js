import React from 'react';
import ArtistQuestionScreen from './artist-question-screen';
import renderer from 'react-test-renderer';

const AVATAR_URL = `https://via.placeholder.com/150/000000/FFFFFF/?text=`;
const artistQuestions = [{
  type: `artist`,
  song: {
    artist: `Пелагея`,
    src: `музыка`,
  },
  answers: [{
    picture: `${AVATAR_URL}Пелагея`,
    artist: `Пелагея`,
  }, {
    picture: `${AVATAR_URL}КДИМБ`,
    artist: `Краснознаменная дивизия имени моей бабушки`,
  }, {
    picture: `${AVATAR_URL}Lorde`,
    artist: `Lorde`,
  }]
}];
const handleAnswer = () => {};

test(`Link changes the class when hovered`, () => {
  const component = renderer.create(
      <ArtistQuestionScreen questions={artistQuestions} getAnswer={handleAnswer} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
