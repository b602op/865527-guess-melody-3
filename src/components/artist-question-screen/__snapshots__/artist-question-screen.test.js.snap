// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<main
  className="app"
>
  <svg
    style={
      Object {
        "left": "-1200em",
        "position": "absolute",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <filter
      id="blur"
    >
      <feGaussianBlur
        in="SourceGraphic"
        stdDeviation="5"
      />
      <feOffset
        dx="0"
        dy="0"
      />
      <feMerge>
        <feMergeNode />
        <feMergeNode
          in="SourceGraphic"
        />
      </feMerge>
    </filter>
  </svg>
  <section
    className="main"
    id="root"
  >
    <section
      className="game game--artist"
    >
      <header
        className="game__header"
      >
        <a
          className="game__back"
          href="#"
          onClick={[Function]}
        >
          <span
            className="visually-hidden"
          >
            Сыграть ещё раз
          </span>
          <img
            alt="Угадай мелодию"
            className="game__logo"
            src="img/melody-logo-ginger.png"
          />
        </a>
        <svg
          className="timer"
          viewBox="0 0 780 780"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            className="timer__line"
            cx="390"
            cy="390"
            r="370"
            style={
              Object {
                "filter": "url(#blur)",
                "transform": "rotate(-90deg) scaleY(-1)",
                "transformOrigin": "center",
              }
            }
          />
        </svg>
        <div
          className="game__mistakes"
        >
          <div
            className="wrong"
          />
          <div
            className="wrong"
          />
          <div
            className="wrong"
          />
        </div>
      </header>
      <section
        className="game__screen"
      >
        <h2
          className="game__title"
        >
          Кто исполняет эту песню?
        </h2>
        <div
          className="game__track"
        >
          <div
            className="track"
          >
            <button
              className="track__button track__button--play"
              type="button"
            />
            <div
              className="track__status"
            >
              <audio />
            </div>
          </div>
        </div>
        <form
          className="game__artist"
        >
          <div
            className="artist"
          >
            <input
              className="artist__input visually-hidden"
              id="answer-1"
              name="answer"
              onClick={[Function]}
              type="radio"
              value="artist-1"
            />
            <label
              className="artist__name"
              htmlFor="answer-1"
            >
              <img
                alt="Пелагея"
                className="artist__picture"
                src="https://via.placeholder.com/150/000000/FFFFFF/?text=Пелагея"
              />
              Пелагея
            </label>
          </div>
          <div
            className="artist"
          >
            <input
              className="artist__input visually-hidden"
              id="answer-2"
              name="answer"
              onClick={[Function]}
              type="radio"
              value="artist-1"
            />
            <label
              className="artist__name"
              htmlFor="answer-2"
            >
              <img
                alt="Краснознаменная дивизия имени моей бабушки"
                className="artist__picture"
                src="https://via.placeholder.com/150/000000/FFFFFF/?text=КДИМБ"
              />
              Краснознаменная дивизия имени моей бабушки
            </label>
          </div>
          <div
            className="artist"
          >
            <input
              className="artist__input visually-hidden"
              id="answer-3"
              name="answer"
              onClick={[Function]}
              type="radio"
              value="artist-1"
            />
            <label
              className="artist__name"
              htmlFor="answer-3"
            >
              <img
                alt="Lorde"
                className="artist__picture"
                src="https://via.placeholder.com/150/000000/FFFFFF/?text=Lorde"
              />
              Lorde
            </label>
          </div>
        </form>
      </section>
    </section>
  </section>
</main>
`;
